packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/model/TextInputNotificationAction.java

```diff
@@ -0,0 +1,63 @@
+package expo.modules.notifications.notifications.model;
+
+import android.content.Context;
+import android.os.Parcel;
+import android.os.Parcelable;
+
+import org.unimodules.core.Promise;
+import org.unimodules.core.arguments.ReadableArguments;
+
+import androidx.annotation.Nullable;
+
+/**
+ * A class representing a single direct reply notification action.
+ */
+public class TextInputNotificationAction extends NotificationAction {
+  private final String mSubmitButtonTitle;
+  private final String mPlaceholder;
+
+  public TextInputNotificationAction(String identifier, String title, boolean opensAppToForeground, String submitButtonTitle, String placeholder) {
+    super(identifier, title, opensAppToForeground);
+    mSubmitButtonTitle = submitButtonTitle;
+    mPlaceholder = placeholder;
+  }
+
+  private TextInputNotificationAction(Parcel in) {
+    super(in);
+    mSubmitButtonTitle = in.readString();
+    mPlaceholder = in.readString();
+  }
+
+  @Override
+  public void writeToParcel(Parcel dest, int flags) {
+    super.writeToParcel(dest, flags);
+    dest.writeString(mSubmitButtonTitle);
+    dest.writeString(mPlaceholder);
+  }
+
+  public static final Creator<TextInputNotificationAction> CREATOR = new Creator<TextInputNotificationAction>() {
+    @Override
+    public TextInputNotificationAction createFromParcel(Parcel in) {
+      return new TextInputNotificationAction(in);
+    }
+
+    @Override
+    public TextInputNotificationAction[] newArray(int size) {
+      return new TextInputNotificationAction[size];
+    }
+  };
+
+  public String getSubmitButtonTitle() {
+    if (mSubmitButtonTitle != null) {
+      return mSubmitButtonTitle;
+    }
+    return super.getTitle();
+  }
+
+  public String getPlaceholder() {
+    if (mPlaceholder != null) {
+      return mPlaceholder;
+    }
+    return "Type something";
```

Comment:
what would happen if none was provided? if it'd look bad could we move that string to android resources so it's translatable?

Reply:
Actually by default it's just blank, which I can see some people wanting. I think returning `null` here is preferable to providing a default